sdk/src/types/block/core/wrapper.rs

```diff
@@ -0,0 +1,348 @@
+
+        let block = Block::unpack::<_, VERIFY>(unpacker, protocol_params)?;
+
+        let signature = Signature::unpack::<_, VERIFY>(unpacker, &())?;
+
+        if VERIFY {
+            let block_len = if let (Some(start), Some(end)) = (start_opt, unpacker.read_bytes()) {
+                end - start
+            } else {
+                block.packed_len()
+            };
+
+            if block_len > Self::LENGTH_MAX {
+                return Err(UnpackError::Packable(Error::InvalidBlockLength(block_len)));
+            }
+        }
+
+        let block_wrapper = Self {
+            protocol_params: protocol_params.clone(),
+            issuing_time,
+            slot_commitment_id,
+            latest_finalized_slot,
+            issuer_id,
+            block,
+            signature,
+        };
+
+        Ok(block_wrapper)
+    }
+}
+
+#[cfg(feature = "serde")]
+pub(crate) mod dto {
+    use serde::{Deserialize, Serialize};
+
+    use super::*;
+    use crate::{
+        types::{block::core::dto::BlockDto, TryFromDto},
+        utils::serde::string,
+    };
+
+    /// The block object that nodes gossip around in the network.
+    #[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+    #[serde(rename_all = "camelCase")]
+    pub struct BlockWrapperDto {
+        pub protocol_version: u8,
+        #[serde(with = "string")]
+        pub network_id: u64,
+        #[serde(with = "string")]
+        pub issuing_time: u64,
+        pub slot_commitment_id: SlotCommitmentId,
+        pub latest_finalized_slot: SlotIndex,
+        pub issuer_id: IssuerId,
+        pub block: BlockDto,
+        pub signature: Signature,
+    }
+
+    impl From<&BlockWrapper> for BlockWrapperDto {
+        fn from(value: &BlockWrapper) -> Self {
+            Self {
+                protocol_version: value.protocol_version(),
+                network_id: value.network_id(),
+                issuing_time: value.issuing_time,
+                slot_commitment_id: value.slot_commitment_id,
+                latest_finalized_slot: value.latest_finalized_slot,
+                issuer_id: value.issuer_id,
+                block: BlockDto::from(&value.block),
+                signature: value.signature,
+            }
+        }
+    }
+
+    impl BlockWrapper {
+        pub fn try_from_dto(dto: BlockWrapperDto, protocol_params: ProtocolParameters) -> Result<Self, Error> {
+            // if dto.protocol_version != protocol_params.version() {
+            //     return Err(Error::ProtocolVersionMismatch {
+            //         expected: protocol_params.version(),
+            //         actual: dto.protocol_version,
+            //     });
+            // }
+
+            // if dto.network_id != protocol_params.network_id() {
+            //     return Err(Error::NetworkIdMismatch {
+            //         expected: protocol_params.network_id(),
+            //         actual: dto.network_id,
+            //     });
+            // }
```

Comment:
Why commented?

Reply:
Will revert, was for a test. But the fact that we have protocol parameters in the wrapper is really annoying, I think we should remove it.

Reply:
Done